Projects/hCore/Sources/PersistencePropertyWrappers.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+@propertyWrapper public struct CachedDefault<T> {
+
+    public let key: String
+    public let defaultValue: T
+
+    public let storage = UserDefaults.standard
+
+    public var wrappedValue: T {
+        get {
+            return storage.object(forKey: key) as? T ?? defaultValue
+        }
+        set {
+            storage.set(newValue, forKey: key)
+            storage.synchronize()
+        }
+    }
+}
+
+@propertyWrapper public struct Cached<T> {
```

Comment:
Niceee! I don't see this used anywhere tho? 😄

Reply:
Going to use it for the `ApplicationState` refactor in to `AppFlow`